### PR TITLE
Add R RCurl code generator

### LINKED ISSUE #347: Add R to Generate Code Snippets
Hi, Please can you add an R code snippet in the Generate Code Snippets section. Thank you for your help.

### PR DESCRIPTION
**Pull request by LuisTejedaS**
Changes
Adds [r-rcurl](https://cran.r-project.org/web/packages/RCurl/index.html) code generation.
Tests ✅

closes #347

### CODE REVIEW

**Review by Carlos-Veloz — APPROVED** [review]

File: codegens/r-rcurl/test/unit/parseBody.test.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  collectionsPath = './fixtures';
+
+describe('parseURLEncodedBody method', function () {
+  it('should return form-url-encoded params', function () {
+    const collection = new sdk.Collection(JSON.parse(
+        fs.readFileSync(path.resolve(__dirname, collectionsPath, './sample_collection.json').toString()))),
+      body = collection.items.members[7].request.body.urlencoded,
+      indentation = '  ',
+      bodyTrim = false,
+      expectedBody = 'c(\n' +
```
Comment (umeshp7) [inline codegens/r-rcurl/test/unit/parseBody.test.js:22]: @LuisTejedaS Why are we adding so many escape characters?

Comment (umeshp7) [reply codegens/r-rcurl/test/unit/parseBody.test.js:22]: Can you post how this would look like when viewed in an editor?

Comment (LuisTejedaS) [reply codegens/r-rcurl/test/unit/parseBody.test.js:22]: Sure @umeshp7  this is the generated snippet with this example, double quotes are preferred in R so here we are escaping them, for this expected value as we are using JS and our preferred are single quotes we have to escape single quotes and we have to scape slashes too, but this is something that we have to do only in our expected JS strings.
Hope this answers the question.

Screen Shot 2022-04-04 at 10 04 16

**Comment by LuisTejedaS** [comment]
Request copy httpMethod:
copyRequest

Request delete httpMethod
DeleteRequest

Request with Follow redirect
followRedirect

Request with disable Query
getRequestWithDisableQuery

Request with link httpMethod
linkRequest

Request with lock httpMethod
lockRequest

Request with multiple headers same name
multipleHeadersSameName

Request with option httpMethod
optionRequest

Request with patch httpMethod
patchRequest

Request post binary file
postBinaryFile

Request post empty form data
postEmptyFormData

Request post form-data special characters
postFormDataSpecialChar

Request post form-data with file
postFromDataWithFile

Request post graphql body json with raw
postGrahpqlBodyJsonWithRaw

Request post javascript with raw
postJavascriptWithRaw

Request post json with raw
postJsonWithRaw

Request post multi part form data with text params
postMultipartFormDataWithTextParam

Request post raw text
postRawText

Request post XML with raw
postTextXMLWithRAw

Request propfind httpMethod
propFindRequest

Request purge httpMethod
purgeRequest


Request put httpMethod
putRequest

Request with headers
RequestHeaders


Request with disabled headers
requestHeadersWithDisabledHeaders

Request with headers with special characters
requestHeadersWithSpecialCharacters

Request with single and multi file upload
singleMultipleFileUploadViaFormData

Request with unlink httpMethod
snippetUnlink

Request with unlock httpMethod
snippetUnlock

Request with view httpMethod
snippetViewRequest

Request with html with raw
textHTMWithRaw

Urlencoded data
urlencodedDataWithDisabledEntries

URL with special characters
urlQuoptesSpecialChar

**Comment by LuisTejedaS** [comment]
> Request copy httpMethod: copyRequest
> 
> Request delete httpMethod DeleteRequest
> 
> Request with Follow redirect followRedirect
> 
> Request with disable Query getRequestWithDisableQuery
> 
> Request with link httpMethod linkRequest
> 
> Request with lock httpMethod lockRequest
> 
> Request with multiple headers same name multipleHeadersSameName
> 
> Request with option httpMethod optionRequest
> 
> Request with patch httpMethod patchRequest
> 
> Request post binary file postBinaryFile
> 
> Request post empty form data postEmptyFormData
> 
> Request post form-data special characters postFormDataSpecialChar
> 
> Request post form-data with file postFromDataWithFile
> 
> Request post graphql body json with raw postGrahpqlBodyJsonWithRaw
> 
> Request post javascript with raw postJavascriptWithRaw
> 
> Request post json with raw postJsonWithRaw
> 
> Request post multi part form data with text params postMultipartFormDataWithTextParam
> 
> Request post raw text postRawText
> 
> Request post XML with raw postTextXMLWithRAw
> 
> Request propfind httpMethod propFindRequest
> 
> Request purge httpMethod purgeRequest
> 
> Request put httpMethod putRequest
> 
> Request with headers RequestHeaders
> 
> Request with disabled headers requestHeadersWithDisabledHeaders
> 
> Request with headers with special characters requestHeadersWithSpecialCharacters
> 
> Request with single and multi file upload singleMultipleFileUploadViaFormData
> 
> Request with unlink httpMethod snippetUnlink
> 
> Request with unlock httpMethod snippetUnlock
> 
> Request with view httpMethod snippetViewRequest
> 
> Request with html with raw textHTMWithRaw
> 
> Urlencoded data urlencodedDataWithDisabledEntries
> 
> URL with special characters urlQuoptesSpecialChar

Hi @umeshp7 here are some generated code snippets for RCurl

**Review by erfemega — APPROVED** [review]

File: codegens/r-rcurl/lib/options.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  return options;
+}
+
+module.exports = {
+  /**
+   * Used in order to get options for generation of PHP-Guzzle code snippet
+   *
+   * @module getOptions
+   *
+   * @returns {Array} Options specific to generation of PHP-Guzzle code snippet
```
[the remark this answers is not in the document]
Comment (LuisTejedaS) [reply codegens/r-rcurl/lib/options.js]: References to guzzle where changed accordingly

**Pull request merged**